src/Functions/FunctionBinaryArithmetic.h

```diff
@@ -246,34 +246,46 @@ enum class OpCase : uint8_t
 constexpr const auto & undec(const auto & x) { return x; }
 constexpr const auto & undec(const is_decimal auto & x) { return x.value; }
 
-template <typename A, typename B, typename Op, typename OpResultType = typename Op::ResultType>
+template <typename A, typename B, typename Op, typename OpResultType = typename Op::ResultType, bool is_compare=false>
 struct BinaryOperation
 {
     using ResultType = OpResultType;
     static const constexpr bool allow_fixed_string = false;
     static const constexpr bool allow_string_integer = false;
 
     template <OpCase op_case>
-    static void NO_INLINE process(const A * __restrict a, const B * __restrict b, ResultType * __restrict c, size_t size, const NullMap * right_nullmap = nullptr)
+    static void NO_INLINE process(const A * __restrict a, const B * __restrict b, ResultType * __restrict c, size_t size, const NullMap * left_nullmap = nullptr, const NullMap * right_nullmap = nullptr, NullMap * result_nullmap = nullptr)
     {
-        if constexpr (op_case == OpCase::RightConstant)
+        if constexpr (!is_compare)
```

Comment:
Please add a comment that states why we need special handling for `least` and `greatest`, otherwise future readers will have no idea.